qiime2_pipeline.bash

```diff
@@ -0,0 +1,169 @@
+#!/bin/bash
+
+if [ $# -ne 4 ]; then
+	echo "Usage: $0 WORK_DIR MAPPING_FP LANE_NUM PROJECT_NAME"
+	exit 1
+fi
+
+WORK_DIR=$1
+MAPPING_FP=$2
+LANE_NUM=$3
+PROJECT_NAME=$4
+
+## GreenGene Training Set artiffact FP
+## This can be downloaded from https://data.qiime2.org/2017.8/common/gg-13-8-99-nb-classifier.qza
+gg_FP=$HOME"/gg-13-8-99-nb-classifier.qza"
+
+DATA_DIR=$WORK_DIR"/data_files"
+
+PROJECT_DIR=$WORK_DIR"/"$PROJECT_NAME
+if [ ! -d $PROJECT_DIR ]; then
+	mkdir $PROJECT_DIR
+fi
+
+FWD=$DATA_DIR"/Undetermined_S0_L$(printf "%03d" $LANE_NUM)_R1_001.fastq.gz"
```

Comment:
I recommend avoiding command substitution inside the string.  Does this work?

    FORMATTED_LANE_NUM=$(printf "%03d" $LANE_NUM)
    FWD="${DATA_DIR}/Undetermined_S0_L${FORMATTED_LANE_NUM}_R1_001.fastq.gz"

Reply:
changed. will test the modified script when it is finalized.